airflow/cli/commands/celery_command.py

```diff
@@ -135,6 +135,7 @@ def worker(args):
 
         ctx = daemon.DaemonContext(
             files_preserve=[handle],
+            umask=cli_utils.get_umask(),
```

Comment:
I think rather than reading from a parent we should allow to set it to a fixed value from configuration (with a reasonable default - for example 0o077). 

From the documentation of umask parameter in DaemonContext:

```
File access creation mask (“umask”) to set for the process on daemon start.

A daemon should not rely on the parent process's umask value, which is beyond its control and may
prevent creating a file with the required access mode. So when the daemon context opens, the umask
is set to an explicit known value.

If the conventional value of 0 is too open, consider setting a value such as 0o022, 0o027, 0o077, or
another specific value. Otherwise, ensure the daemon creates every file with an explicit access mode
for the purpose.
```

via https://kite.com/python/docs/daemon.DaemonContext

Reply:
I have referred few articles to get into the depth but could find why the value should not depend on parent process or what repercussions can it have if I inherit it from parent process . However all articles suggest it to set to a predefined value. I will make the value configurable with a reasonable default. 

- https://smnd.sk/anino/programming/c/c-faq_comp.unix.programmer/faq_2.html
- http://www.enderunix.org/docs/eng/daemon.php

Reply:
The problem is that you do not know what the umask is in the parent process - it depends on the system umask setting. It's a nuance but I think it's a good practice to have it fixed.

Reply:
Just to give some example - there are problems with Dockerfile caching when git by default uses umask of the systems (for example debian/mint had it different and different versions) rather than have predefined value. Different systems at different times had different default umask and that means that your daemon might behave differently for those different systems,

Reply:
Thanks for explanation. Things are much clearer to me now!

Reply:
I have incorporated the changes as suggested. Umask is read from configuration. Also updated corresponding docs and configuration file.